### PR TITLE
NIFI-2604: Added validators & logic for multiple URLs/files/folders for DB driver location

### PR DESCRIPTION
**Pull request by mattyb149**

### CODE REVIEW

File: nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/StandardValidators.java [hunk 1]
```diff
@@ -379,6 +381,68 @@ public ValidationResult validate(final String subject, final String input, final
...
+            if (context.isExpressionLanguageSupported(subject) && context.isExpressionLanguagePresent(input)) {
+                return new ValidationResult.Builder().subject(subject).input(input).explanation("Expression Language Present").valid(true).build();
+            }
+            try {
+                if (input == null) {
+                    return new ValidationResult.Builder().subject(subject).input(null).explanation("List must have at least one non-empty element").valid(false).build();
+                }
+                final String[] list = input.split(",");
+                for (String item : list) {
+                    String itemToValidate = (StringUtils.isBlank(item) && trimEntries) ? item.trim() : item;
```
Comment (brosander) [inline nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/StandardValidators.java]: Why do we only trim if the string is blank?

Shouldn't we trim anytime trimEntries is true?

File: nifi-commons/nifi-processor-utilities/src/test/java/org/apache/nifi/processor/util/TestStandardValidators.java [hunk 2]
```diff
@@ -100,4 +100,105 @@ public void testDataSizeBoundsValidator() {
         vr = val.validate("DataSizeBounds", "water", validationContext);
         assertFalse(vr.isValid());
     }
+
+    @Test
+    public void testListValidator() {
+        Validator val = StandardValidators.createListValidator(true, false, StandardValidators.NON_EMPTY_VALIDATOR);
```
Comment (brosander) [inline nifi-commons/nifi-processor-utilities/src/test/java/org/apache/nifi/processor/util/TestStandardValidators.java]: The trimming behavior would probably be easier to test with the BOOLEAN_VALIDATOR or similar.

Ex:
```
vr = val.validate("List", "notbool", validationContext);
assertFalse(vr.isValid());

vr = val.validate("List", "true", validationContext);
assertTrue(vr.isValid());

vr = val.validate("List", "    true   \n   ", validationContext);
assertTrue(vr.isValid());
```

File: nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/StandardValidators.java (hunk 1 above)
Comment (mattyb149) [inline nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/StandardValidators.java]: Good catch, will fix

File: nifi-commons/nifi-processor-utilities/src/test/java/org/apache/nifi/processor/util/TestStandardValidators.java (hunk 2 above)
Comment (mattyb149) [inline nifi-commons/nifi-processor-utilities/src/test/java/org/apache/nifi/processor/util/TestStandardValidators.java]: Good idea, will incorporate

**Comment by brosander** [comment]
+1 contingent on trim fixes and more tests with extra whitespace around values

**Comment by JPercivall** [comment]
+1

Looks good @mattyb149, I will merge it in. Thanks @brosander for helping out with the review.
